### PR TITLE
Add config store support to Threshold Detection Agent

### PR DESCRIPTION
**Pull request by mikeroup**
#1057 #982

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/1064)

### CODE REVIEW

File: services/core/ThresholdDetectionAgent/thresholddetection/agent.py [hunk 1]
```diff
@@ -66,20 +66,19 @@
...
 _log = logging.getLogger(__name__)
-__version__ = '3.6'
+__version__ = '3.7'
 
 
 def thresholddetection_agent(config_path, **kwargs):
     """Load configuration for ThresholdDetectionAgent
 
-    :param config_path: Path to a configuration file.
+    :param config_path: Path to a configuration file. Ignored.
```
Comment (craig8) [inline services/core/ThresholdDetectionAgent/thresholddetection/agent.py]: Shouldn't ignore, we should instead put them as defaults. and use the set_default function of the config store.  See actuator for example

**Comment by mikeroup** [comment]
Considering removing the "message" config fields completely and generating it inside of the agent. Thoughts?

**Comment by mikeroup** [comment]
@craig8 please look at this again

File: services/core/ThresholdDetectionAgent/thresholddetection/agent.py (hunk 1 above)
Comment (craig8) [inline services/core/ThresholdDetectionAgent/thresholddetection/agent.py]: You didn't address this

Comment (mikeroup) [inline services/core/ThresholdDetectionAgent/thresholddetection/agent.py]: I set the default config in line 118

Comment (craig8) [inline services/core/ThresholdDetectionAgent/thresholddetection/agent.py]: Then why is the comment set as Ignored if you load from the config file?

Comment (mikeroup) [inline services/core/ThresholdDetectionAgent/thresholddetection/agent.py]: That's not there anymore. Which commit are you looking at?

Comment (craig8) [inline services/core/ThresholdDetectionAgent/thresholddetection/agent.py]: https://github.com/VOLTTRON/volttron/pull/1064/files/a59aec394e6d48370e5e80137643cfe3f225666a
The view shows that it is there.

**Comment by mikeroup** [comment]
Working on docs now

File: docs/source/core_services/service_agents/threshold/ThresholdAgent.rst [hunk 2]
```diff
@@ -0,0 +1,39 @@
+.. _ThresholdAgent:
+
+==========================
+Threshold Detection Agent
+==========================
+
+The ThresholdDetectionAgent will publish an alert when a value published to a
+topic exceeds a configured value. The agent can be configured to watch topics
+are associated with a single value or to watch devices' all topics.
```
Comment (craig8) [inline docs/source/core_services/service_agents/threshold/ThresholdAgent.rst]: Exceeds only?

File: docs/source/core_services/service_agents/threshold/ThresholdAgent.rst [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+Configuration
+-------------
+
+The ThresholdDetectionAgent supports the configstore and can be configured
+with a file named "config".
```
Comment (craig8) [inline docs/source/core_services/service_agents/threshold/ThresholdAgent.rst]: Add a link to the configstore documentation

**Pull request merged**